agent/tests/test_user_instrument.c

```diff
@@ -145,6 +145,32 @@ static void test_hashmap_wraprec() {
 }
 #endif /* PHP >= 7.4 */
 
+static void test_add_custom_tracer_named() {
+  nruserfn_t* wr = NULL;
+  tlib_php_request_start();
+
+  wr = nr_php_add_custom_tracer_named(NULL, 10);
+  tlib_pass_if_null("add_custom_tracer_named with NULL name", wr);
+
+  wr = nr_php_add_custom_tracer_named("function_name", 0);
+  tlib_pass_if_null("add_custom_tracer_named with length == 0", wr);
+
+  // nr_php_add_custom_tracer_named's second argument is of size_t type,
+  // which is **unsigned**!!! However, under the hood, it calls a function
+  // that converts it to an int! So this is testing that no unwanted
+  // side effects will happen when this silent type conversion takes place.
+  wr = nr_php_add_custom_tracer_named("function_name", -1);
+  tlib_pass_if_null("add_custom_tracer_named with length < 0", wr);
+
+  wr = nr_php_add_custom_tracer_named(NR_PSTR("scope_name::"));
+  tlib_pass_if_null("add_custom_tracer_named with name ending with :", wr);
+
+  wr = nr_php_add_custom_tracer_named(NR_PSTR("scope_name::function_name"));
+  tlib_pass_if_not_null("add_custom_tracer_named with valid name", wr);
+
```

Comment:
Thanks for adding the tests for add_custom_tracer_named.
A test for the following would be good for completeness:
```
/* Valid func name, no scope */
  wr = nr_php_add_custom_tracer_named(NR_PSTR("function_name"));
  tlib_pass_if_not_null("add_custom_tracer_named with valid func name, no scope", wr);
```

Reply:
See 9748cf5649dc3f3e179ae74842ede3a1a03119d8.